pkg/operator/revisioncontroller/revision_controller.go

```diff
@@ -366,6 +366,28 @@ func (c RevisionController) sync(ctx context.Context, syncCtx factory.SyncContex
 		return nil
 	}
 
+	// If there is a revision available, make sure that revision-ready is set to true on the
+	// revision-status{status.latestAvailableRevision} configmap. This prevents
+	// blocking the controller when upgrading to a version that checks revision
+	// readiness.
+	if operatorStatus.LatestAvailableRevision != 0 {
+		configMap, err := c.configMapGetter.ConfigMaps(c.targetNamespace).Get(ctx, nameFor("revision-status", operatorStatus.LatestAvailableRevision), metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+
+		if configMap.Annotations == nil {
+			configMap.Annotations = make(map[string]string)
```

Comment:
Is this mutating an object in the informer cache?

Reply:
not obvious :(